### PR TITLE
create dataset on push

### PR DESCRIPTION
**Pull request by moshe-kabala**
- Add SetDatasetConfig
- Add library for user interactions
- Add function that add dataset if not exists

### CODE REVIEW

File: cmd/datasets/push.go [hunk 1]
```diff
@@ -81,3 +89,43 @@ func getDatasetFiles(datasetConfig *config.DatasetConfig) ([]string, error) {
 	}
 	return allMatchedFiles, nil
 }
+
+func addDatasetIfNotExisted(ctx context.Context, datasetConfig *config.DatasetConfig) error {
+	datasets, _, err := ApiClient.GetDatasets(ctx).Execute()
+	if err != nil {
+		return fmt.Errorf("Failed to get datasets: %v", err)
```
Comment (YuvalFeldman) [inline cmd/datasets/push.go:96]: use logger class for this as well?

File: cmd/datasets/push.go [hunk 2]
```diff
@@ -81,3 +89,43 @@ func getDatasetFiles(datasetConfig *config.DatasetConfig) ([]string, error) {
...
+	isDatasetExisted := false
+	for _, dataset := range datasets.Datasets {
+		if dataset.Cid == datasetConfig.DatasetId {
+			isDatasetExisted = true
+			break
+		}
+	}
+
+	if isDatasetExisted {
+		log.Infof("Not found dataset id: %s, will create a new one", datasetConfig.DatasetId)
```
Comment (YuvalFeldman) [inline cmd/datasets/push.go:107]:
```suggestion
		log.Infof("Not found dataset id: %s. Creating new dataset", datasetConfig.DatasetId)
```

File: cmd/datasets/push.go (hunk 1 above)
Comment (YuvalFeldman) [reply cmd/datasets/push.go:96]: same for the other usages, I wont comment on each

**Review by YuvalFeldman — APPROVED** [review]
LGTM, See comments

File: cmd/datasets/push.go (hunk 1 above)
Comment (moshe-kabala) [reply cmd/datasets/push.go:96]: fmt.Errorf not Print error (It create a error)

File: cmd/datasets/push.go (hunk 2 above)
Comment (moshe-kabala) [reply cmd/datasets/push.go:107]: fixed

**Pull request merged**